src/graph/impl/KokkosGraph_GraphColor_impl.hpp

```diff
@@ -2379,6 +2379,234 @@ class GraphColor_VB:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t
 };  // class GraphColor_VB
 
 
+
+
+/*! \brief Class for the deterministic vertex based graph coloring algorithms.
+ */
+template <typename HandleType, typename lno_row_view_t_, typename lno_nnz_view_t_>
+class GraphColor_VBD:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t_>{
+public:
+
+  typedef long long int ban_type;
+
+  typedef lno_row_view_t_ in_lno_row_view_t;
+  typedef lno_nnz_view_t_ in_lno_nnz_view_t;
+  typedef typename HandleType::color_view_t color_view_type;
+
+
+  typedef typename HandleType::size_type size_type;
+  typedef typename lno_row_view_t_::device_type row_lno_view_device_t;
+
+  typedef typename HandleType::nnz_lno_t nnz_lno_t;
+
+  typedef typename HandleType::color_t color_t;
+  typedef typename HandleType::color_host_view_t color_host_view_t; //Host view type
+
+  typedef typename HandleType::HandleExecSpace MyExecSpace;
+  typedef typename HandleType::HandleTempMemorySpace MyTempMemorySpace;
+  typedef typename HandleType::HandlePersistentMemorySpace MyPersistentMemorySpace;
+
+  typedef typename Kokkos::View<nnz_lno_t, row_lno_view_device_t> single_dim_index_view_type;
+  typedef typename single_dim_index_view_type::HostMirror single_dim_index_host_view_type; //Host view type
+
+  typedef Kokkos::RangePolicy<MyExecSpace> my_exec_space;
+
+  typedef typename HandleType::size_type_temp_work_view_t size_type_temp_work_view_t;
+  typedef typename HandleType::size_type_persistent_work_view_t size_type_persistent_work_view_t;
+
+
+  typedef typename HandleType::nnz_lno_temp_work_view_t nnz_lno_temp_work_view_t;
+  typedef typename HandleType::nnz_lno_persistent_work_view_t nnz_lno_persistent_work_view_t;
+
+
+
+  typedef typename in_lno_row_view_t::const_type const_lno_row_view_t;
+
+
+  typedef typename lno_nnz_view_t_::const_type const_lno_nnz_view_t;
+  typedef typename lno_nnz_view_t_::non_const_type non_const_lno_nnz_view_t;
+
+
+protected:
+
+
+
+  bool _ticToc; //if true print info in each step
+  int _chunkSize; //the size of the minimum work unit assigned to threads. Changes the convergence on GPUs
+  char _use_color_set; //the VBD algorithm type.
```

Comment:
I thought this sort of this would be in the handle ..

Reply:
For some reason it is being kept separate in the other GraphColor classes, so I am using the same pattern. I could look into why/if this design is appropriate?